### ISSUE TITLE
[Question] Why 256 Epoch for German and 512 for Russian in Transfer learning (Based on Paper)?

### ISSUE THREAD

**Issue by soheiltehranipour** [NVIDIA/NeMo#1443]
Question
First of all, thank you guys specially @okuchaiev for this magnificent project on Conversational A.I. Good job Guys.

I am using NeMo **ASR** for **Persian** language mainly because of being able to use **transfer learning** due to lack of labeled data. I got 40% WER on my validation data after working on NeMo for about 3 months. For more information, I am using Quartznet15x5 (En) and used transfer learning for another language. _But there is 2 main question here:_

1. According to the [Paper ](https://arxiv.org/abs/2005.04290): "The Russian model was trained for 512 epochs while the Spanish and German ones for 256 epochs."
But I am stuck on the 40% WER after almost 15th epoch and it is not changing even after 50 epochs. **Am i experiencing overfitting here?**
On the other hand, based on the main use of transfer learning, there is no need to train the per-trained model for this **such large number of epochs.** Please let me know what is the reason of this strategy of training.

2. **If this is overfitting, How can i regularize it?** 
(Can Dropout help me with that? ) If YES, is there any advice for setting that?

Environment overview:
-Environment location: Jupyter Notebook (Cloud Services)
-Method of NeMo install: !python -m pip install git+https://github.com/NVIDIA/NeMo.git@main#egg=nemo_toolkit[all]

Environment details
-OS version : Ubuntu 18.04
-PyTorch version : 1.6.0+cu101
-Python version : Python 3.6.9
-GPU model: GTX 1080 Ti

**Comment by rbracco** [comment]
Can you share more info about your training process?

-How many hours of training data do you have?
-What is the size of the Persian vocab? Did you change the decoder to use the new vocab? 
-What is the loss doing? Is it falling to a certain level and stalling? What's the lowest loss you get and can you share a graph if possible? 

If you have a colab notebook I'd be happy to take a look. I also discovered a possible flaw in the initialization of the decoder weights in NeMo as discussed in #1462. Using the standard initialization my transfer learning frequently fails to converge. 

I had more success by doing the following:

```
quartznet = nemo_asr.models.EncDecCTCModel.from_pretrained(model_name="QuartzNet15x5Base-En")
quartznet.change_vocabulary(new_vocabulary=en_ipa_vocab)
print(quartznet.decoder)
```

![92c750a436cccb6429eba20cdf16f5a9](https://user-images.githubusercontent.com/47190785/99440937-3d89e080-28e5-11eb-9c62-d103db67aeab.png)


Then manually reinitialize the first layer of the decoder so that it uses PyTorch's default Kaiming Uniform initialization instead of NeMo's Xavier Uniform. Replace <N_CLASSES> with the number of symbols in your vocab (+1 for the blank token). It is the same value that has a red box around it in my decoder above.  

```
quartznet.decoder.decoder_layers[0] = nn.Conv1d(1024, <N_CLASSES>, kernel_size=1, stride=1)
```

Hope this helps and let me know if you have any questions.
